helm-base-versions/send-json-to-cosmos.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+set -e
+
+COSMOS_KEY=$1
+CHART_NAME=$2
+NAMESPACE=$3
+CLUSTER_NAME=$4
+DEPRECATED_CHART_NAME=$5
+CURRENT_VERSION=$6
+IS_DEPRECATED=$7
+IS_ERROR=$8
+
+#Define COSMOS variables
+VERB="POST"
+RESOURCE_TYPE="docs"
+RESOURCE_ID="dbs/platform-metrics/colls/app-helm-chart-metrics"
+RESOURCE_LINK="$RESOURCE_ID/$RESOURCE_TYPE"
+COSMOS_URL="https://pipeline-metrics.documents.azure.com/$RESOURCE_LINK"
+NOW=$(TZ=GMT date '+%a, %d %b %Y %T %Z')
+
+SIGNATURE="$(printf "%s" "$VERB\n$RESOURCE_TYPE\n$RESOURCE_ID\n$NOW" | tr '[A-Z]' '[a-z]')\n\n"
+HEX_KEY=$(printf "$COSMOS_KEY" | base64 --decode | hexdump -v -e '/1 "%02x"')
+hashedSignature=$(printf "$SIGNATURE" | openssl dgst -sha256 -mac hmac -macopt hexkey:$HEX_KEY -binary | base64)
+AUTH_STRING="type=master&ver=1.0&sig=$hashedSignature"
+URL_ENCODED_STRING=$(printf "$AUTH_STRING" | sed 's/=/%3d/g' | sed 's/&/%26/g' | sed 's/+/%2b/g' | sed 's/\//%2f/g')
+
```

Comment:
I suspect it would be safer to use an SDK via node.js or python.

there's a risk here that the bash tools won't handle some edge cases that the SDKs / a higher level language have automated test coverage for and would be easier to understand.